plugins/by-name/nvim-ufo/default.nix

```diff
@@ -1,87 +1,133 @@
 {
   lib,
-  helpers,
-  config,
-  pkgs,
   ...
 }:
-with lib;
 let
-  cfg = config.plugins.nvim-ufo;
+  inherit (lib.nixvim) defaultNullOpts;
 in
-{
-  options.plugins.nvim-ufo = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "nvim-ufo";
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "nvim-ufo";
+  luaName = "ufo";
+  package = "nvim-ufo";
 
-    package = lib.mkPackageOption pkgs "nvim-ufo" {
-      default = [
-        "vimPlugins"
-        "nvim-ufo"
-      ];
-    };
+  maintainers = [ lib.maintainers.khaneliman ];
 
-    openFoldHlTimeout = helpers.defaultNullOpts.mkInt 400 ''
+  # TODO: added 2024-10-05 remove after 24.11
+  deprecateExtraOptions = true;
+  optionsRenamedToSettings = [
+    "openFoldHlTimeout"
+    "providerSelector"
+    "closeFoldKinds"
+    "foldVirtTextHandler"
+    "enableGetFoldVirtText"
+    [
+      "preview"
+      "winConfig"
+      "border"
+    ]
+    [
+      "preview"
+      "winConfig"
+      "winblend"
+    ]
+    [
+      "preview"
+      "winConfig"
+      "winhighlight"
+    ]
+    [
+      "preview"
+      "winConfig"
+      "maxheight"
+    ]
+    [
+      "preview"
+      "mappings"
+    ]
+  ];
+
+  settingsOptions = {
+    open_fold_hl_timeout = defaultNullOpts.mkInt 400 ''
```

Comment:
```suggestion
    open_fold_hl_timeout = defaultNullOpts.mkUnsignedInt 400 ''
```

Reply:
Wondering if the winblend and maxheight should be, as well..

Reply:
Updated them both, winblend is 0-100 and maxheight doesn't sound like something that could be negative.